### PR TITLE
Add TypeDoc support 

### PR DESCRIPTION
**Pull request by James-Oswald**
I added JSDoc as an NPM package

### CODE REVIEW

**Review by 10dine — COMMENTED** [review]
This might be automated code but I think jsdoc should be in devdependencies. It doesn't seem like a module that would be needed in prod.

**Comment by James-Oswald** [comment]
> This might be automated code but I think jsdoc should be in devdependencies. It doesn't seem like a module that would be needed in prod.

I can't believe you actually caught this lol

**Comment by 10dine** [comment]
Consider your code reviewed.

**Review by 10dine — CHANGES_REQUESTED** [review]
Remember to actually change it and put JSDocs in devdependencies.

**Comment by James-Oswald** [comment]
Note, we have to use [Typedoc](https://typedoc.org/) instead of jsdoc since it turns out jsdoc does not support typescript as i originally thought it did.

File: typedoc.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+    "$schema": "https://typedoc.org/schema.json",
+    "readme": "README.md",
+    "includeVersion": true,
+    "entryPointStrategy": "expand",
+    "entryPoints": ["src/AEG"], //This needs to be changed once more classes are added outside AEG
```
Comment (10dine) [inline typedoc.json:6]: I'd suggest pinning this in discord. Or make an important pull request checklist channel and put it there. Unless you don't foresee that many classes being added in the future.

**Review by 10dine — COMMENTED** [review]
Looks good. Though PR should be a bit more specific. For example, should point out you changed the "build" action to "lint", added a whole new action named Validate Documentation Build, and you may also add that you changed some basic documentation such as authors in some files.

**Pull request merged**